JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -121,71 +113,64 @@ fun HomeRoute(
-                check(uiState is HomeUiState.HasPosts)
+    // Modifier to be applied to the all HomeScreen types
+    val screenModifier = Modifier.homeScreenPadding(showTopAppBar)
+    val homeScreenType = getHomeScreenType(showTopAppBar, uiState)
+    when (homeScreenType) {
+        HomeScreenType.FeedWithArticleDetails -> {
+            HomeFeedWithArticleDetailsScreen(
+                uiState = uiState,
+                showTopAppBar = showTopAppBar,
+                onToggleFavorite = onToggleFavorite,
+                onSelectPost = onSelectPost,
+                onRefreshPosts = onRefreshPosts,
+                onErrorDismiss = onErrorDismiss,
+                onInteractWithList = onInteractWithFeed,
+                onInteractWithDetail = onInteractWithArticleDetails,
+                openDrawer = openDrawer,
+                homeListLazyListState = homeListLazyListState,
+                articleDetailLazyListStates = articleDetailLazyListStates,
+                scaffoldState = scaffoldState,
+                modifier = screenModifier
+            )
+        }
+        HomeScreenType.Feed -> {
+            HomeFeedScreen(
+                uiState = uiState,
+                showTopAppBar = showTopAppBar,
+                onToggleFavorite = onToggleFavorite,
+                onSelectPost = onSelectPost,
+                onRefreshPosts = onRefreshPosts,
+                onErrorDismiss = onErrorDismiss,
+                openDrawer = openDrawer,
+                homeListLazyListState = homeListLazyListState,
+                scaffoldState = scaffoldState,
+                modifier = screenModifier
+            )
+        }
+        HomeScreenType.ArticleDetails -> {
+            // Guaranteed by above condition for home screen type
+            check(uiState is HomeUiState.HasPosts)
 
-                ArticleScreen(
-                    post = uiState.selectedPost,
-                    showNavRail = showNavRail,
-                    onBack = onInteractWithFeed,
-                    isFavorite = uiState.favorites.contains(uiState.selectedPost.id),
-                    onToggleFavorite = {
-                        onToggleFavorite(uiState.selectedPost.id)
-                    },
-                    lazyListState = articleDetailLazyListStates.getValue(
-                        uiState.selectedPost.id
-                    ),
-                    modifier = screenModifier
-                )
+            ArticleScreen(
+                post = uiState.selectedPost,
+                showNavRail = !showTopAppBar,
+                onBack = onInteractWithFeed,
+                isFavorite = uiState.favorites.contains(uiState.selectedPost.id),
+                onToggleFavorite = {
+                    onToggleFavorite(uiState.selectedPost.id)
+                },
+                lazyListState = articleDetailLazyListStates.getValue(
+                    uiState.selectedPost.id
+                ),
+                modifier = screenModifier
```

Comment:
On the article screen when there is no nav bar, the navigation bar insets are getting applied twice right now:
![device-2021-09-17-141118](https://user-images.githubusercontent.com/4217560/133854086-6646f84e-32a1-47b5-b386-f9eefe2b5173.png)